blob/gcsblob/gcsblob.go

```diff
@@ -160,15 +180,19 @@ func (o *URLOpener) forParams(ctx context.Context, q url.Values) (*Options, erro
 	}
 	opts := new(Options)
 	*opts = o.Options
-	if accessID := q.Get("access_id"); accessID != "" {
+	if accessID := q.Get("access_id"); accessID != "" && accessID != opts.GoogleAccessID {
 		opts.GoogleAccessID = accessID
+		opts.PrivateKey = nil // Clear any previous key unrelated to the new accessID.
 	}
 	if keyPath := q.Get("private_key_path"); keyPath != "" {
 		pk, err := ioutil.ReadFile(keyPath)
 		if err != nil {
 			return nil, err
 		}
 		opts.PrivateKey = pk
+	} else if _, exists := q["private_key_path"]; exists {
```

Comment:
What's this for?

Reply:
Tests and to force going through the *Credentials API*. I'll add that to the comment.